x/twap/types/keys.go

```diff
@@ -66,12 +67,33 @@ func FormatHistoricalTimeIndexTWAPKey(accumulatorWriteTime time.Time, poolId uin
 }
 
 func FormatHistoricalPoolIndexTWAPKey(poolId uint64, denom1, denom2 string, accumulatorWriteTime time.Time) []byte {
-	var buffer bytes.Buffer
 	timeS := osmoutils.FormatTimeString(accumulatorWriteTime)
-	fmt.Fprintf(&buffer, "%s%d%s%s%s%s%s%s", HistoricalTWAPPoolIndexPrefix, poolId, KeySeparator, denom1, KeySeparator, denom2, KeySeparator, timeS)
+	return FormatHistoricalPoolIndexTWAPKeyFromStrTime(poolId, denom1, denom2, timeS)
+}
+
+func FormatHistoricalPoolIndexTWAPKeyFromStrTime(poolId uint64, denom1, denom2 string, accumulatorWriteTimeString string) []byte {
+	var buffer bytes.Buffer
+	fmt.Fprintf(&buffer, "%s%d%s%s%s%s%s%s", HistoricalTWAPPoolIndexPrefix, poolId, KeySeparator, denom1, KeySeparator, denom2, KeySeparator, accumulatorWriteTimeString)
```

Comment:
Not relevant for this PR, but we may want to have a generic `osmoutils.BuildKey(elements ...interface{})` that does this to avoid potential future key issues

Reply:
also, should this end un a `KeySeparator`? Right now we're not appending anything, but if someone adds a traling key part in the future it may become an issue

```suggestion
	fmt.Fprintf(&buffer, "%s%d%s%s%s%s%s%s%s", HistoricalTWAPPoolIndexPrefix, poolId, KeySeparator, denom1, KeySeparator, denom2, KeySeparator, accumulatorWriteTimeString, KeySeparator)
```

Reply:
actually, ignore that. It would require a migration of existing keys.

Reply:
I think your right, alas wasn't thought about before. would take a migration now